python-lib/lal/handlers/lal_handler.py

```diff
@@ -44,9 +61,22 @@ def get_meta_by_status(self, status=None):
                 (self.meta_df.status == status) & (self.meta_df.annotator == self.current_user)]
 
     def get_remaining(self):
-        seen_ids = set(self.get_meta_by_status().data_id.values)
-        logging.info("get_remaining: Seen ids: {0}".format(seen_ids))
-        return [i for i in self.classifier.get_all_item_ids_list() if i not in seen_ids]
+        labeled_ids = set(self.get_meta_by_status().data_id.values)
+        logging.info("get_remaining: Seen ids: {0}".format(labeled_ids))
+        unlabeled_ids = [i for i in self.classifier.get_all_item_ids_list() if i not in labeled_ids]
+        result = []
+        with LALHandler.lock:
+            for i in unlabeled_ids:
+                if i in self.sample_by_user_reservation:
+                    reserved_sample = self.sample_by_user_reservation.get(i)
+                    if reserved_sample.reserved_until <= datetime.now():
+                        del self.sample_by_user_reservation[i]
+                    else:
+                        if reserved_sample.username == self.current_user:
+                            result.append(i)
```

Comment:
We might also consider resetting the `reserved_until` (i.e. removing the `i` from the reserved list) so that the images are locked for the same amount of time.

Reply:
Not sure I understand what you mean by locking images for the same amount of time

Reply:
We had a discussion to clarify with @andreybavt and the append was added. So => resolved.